crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -294,6 +307,51 @@ async fn handle_coinbase(state: StateType) -> ResponseData<Address> {
     }
 }
 
+async fn handle_evm_increase_time(
+    state: StateType,
+    increment: U256OrUsize,
+) -> ResponseData<String> {
+    event!(Level::INFO, "evm_increaseTime({increment:?})");
+    let increment: U256 = increment.into();
+    let mut offset = state.block_time_offset_seconds.write().await;
+    *offset += increment;
+    ResponseData::Success {
+        result: offset.to_string(),
+    }
+}
+
+async fn handle_evm_set_next_block_timestamp(
+    state: StateType,
+    timestamp: U256OrUsize,
+) -> ResponseData<String> {
+    event!(Level::INFO, "evm_setNextBlockTimestamp({timestamp:?})");
+    match state.blockchain.read().await.last_block() {
+        Ok(latest_block) => {
+            match Into::<U256>::into(timestamp.clone()).checked_sub(latest_block.header.timestamp) {
```

Comment:
```suggestion
            match U256::from(timestamp.clone()).checked_sub(latest_block.header.timestamp) {
```

Reply:
I had tried it this way at first, but I didn't like the result. It gives this error:
![image](https://github.com/NomicFoundation/hardhat/assets/7883777/b75c1465-8c19-41a1-9494-0d3f99ea065b)
If I understand correctly, this is because the `ruint` code is set up to generically require a `try_from` for _any_ conversion from an integer. Even though I code it with `::from(_)`, not `::try_from(_)`, it still complains about wanting a `TryFrom`. That's a little quirky, isn't it? Or am I missing something?

In any case, if I'm following the code correctly, the only type of error that could occur is `FromUintError::Overflow`, aka `The Uint value is too large for the target type`. And, in this case, I _know_ that `timestamp` will fit into a `U256`, so I wanted to avoid implementing `TryFrom<U256OrUsize>`.

I'm happy to code this however you'd like, just let me know, but I thought I'd explain first that I _did_ try it your suggested way first, and to explain my rationale for why I settled on the way you see it written here.

Reply:
My understanding of the error is that `From` is not implemented for the `timestamp` type. Is it a `U256OrSize`?

Reply:
Yes, it's a `U256OrSize`, and I actually did already `impl From<U256OrUsize> for U256` alongside `enum U256OrUsize`.

Note that, at least in my IDE, jumping to the definition of `U256::from()` goes to a method on `impl<...> Uint`, not to an implementation of `From<_>`, and that implementation simply delegates to `UintTryFrom`, panicking upon failure. I'm fairly confident that the error is that `TryFrom<U256OrUsize>` is the complaint here.

I went ahead and implemented `TryFrom<U256OrUsize> for U256`, and was able to convert this into `U256::from(_)` (and was required to convert a few other places too, since it told me that I couldn't _also_ implement `From<U256OrUsize>` due to a conflict). I did that in [this commit](https://github.com/NomicFoundation/hardhat/pull/4227/commits/14e602714da7bd480d2a1071990c568ef0325811). It had seemed counterintuitive to me to implement `TryFrom` when I know for sure that converting a `u64` to a `U256` will _never_ induce an `Overflow` error, but if you think the code is cleaner this way then I'm good with that. Let me know if you like that or if you'd like me to revert it so we can do something different.

Reply:
That's very odd. I can only imagine that for some reason `U256` doesn't have the conversion version for `usize` because I've used the pattern `U256::from(value)` for a `u64`, IIRC.

Feel free to revert it to what you had originally then, with the `Into::into` and leave it at that. That prevents the unnecessary creation of an error path